core/src/main/java/tc/oc/pgm/flag/Flag.java

```diff
@@ -264,27 +278,48 @@ public boolean canDropAt(Location location) {
   private int sequentialPostCounter = 1;
 
   public Post getReturnPost(Post post) {
```

Comment:
I want to point out how overly complex this function has become, it used to be simply:
```
  public Location getReturnPoint(Post post) {
    return post.getReturnPoint(this, this.bannerYawProvider).clone();
  }
```

Now there's `predeterminePost`, `getReturnPoint`, `getReturnPost`, `getBannerYawProvider` to accomplish the same goal, and with this PR it only gets even increasingly more complicated.

This is screaming for a need of simplification.

Showing the next post when the flag is picked up on the carried state is the only thing that needs `predeterminePost` which may be completely incompatible with filters, you can't know what the filter output may say when it's evaluated later, and the filter should only be evaluated when flag needs to decide the next post.